framework/src/media/FileInputDataSource.cpp

```diff
@@ -111,72 +136,35 @@ bool FileInputDataSource::isPrepare()
 	return true;
 }
 
-ssize_t FileInputDataSource::read(unsigned char *buf, size_t size)
+// ???
+int FileInputDataSource::readAt(long offset, int origin, unsigned char *buf, size_t size)
 {
-	size_t rlen = 0;
-	size_t readSize = size;
-	std::shared_ptr<Decoder> decoder = getDecoder();
-
-	if (!buf) {
-		meddbg("buf is nullptr, hence return EOF\n");
-		return EOF;
+	if (fseek(mFp, offset, origin) != 0) {
+		meddbg("FileInputDataSource::readAt : fail to seek\n");
+		return -1;
 	}
 
-	if (decoder) {
-		size_t frames = size;
-		rlen = getDecodeFrames(buf, &frames);
-		medvdbg("decode frame : %d/%d\n", rlen, size);
-		if (rlen == size) {
-			return rlen;
-		}
-
-		/* Set buf offset */
-		buf += rlen;
-		/* Calculate available space in 'buf' */
-		readSize -= rlen;
+	return fread(buf, sizeof(unsigned char), size, mFp);
+}
 
-		if (readSize > decoder->getAvailSpace()) {
-			readSize = decoder->getAvailSpace();
-		}
-	}
+ssize_t FileInputDataSource::onStreamBufferWritable()
+{
+	unsigned char buf[1024];
```

Comment:
Please find a way to use stream buffer directly.

Reply:
File raw data need to be decoded, and then push to stream buffer...
I'm afraid we can not reuse stream buffer here.